app/models/computer.rb

```diff
@@ -0,0 +1,38 @@
+class Computer < ActiveRecord::Base
+  has_paper_trail class_name: 'Version'
+
+  belongs_to :country, required:false
+  has_one :package, as: :detail, dependent: :destroy
+  after_create :create_on_stockit
+  after_update :update_on_stockit
+  before_save :downcase_brand, if: :brand_changed?
+  before_save :set_updated_by
+
+  private
+
+  def create_on_stockit
```

Comment:
Given that there are a few classes which are very similar, might be worth moving some of this code into a concern. Would make it easier to manage and to add more of those detail models in the future : )

Reply:
Yes I was also thinking of something like that earlier, creating a concern that would cater all three types. Also in future if we want to add any similar class, we can just include that concern and perform same operations on the record.